### DISCUSSION TITLE
Slow Animations and Disappearing DOM Elements on Mobile Scroll with Nuxt 3 Migration

### DISCUSSION THREAD

**Discussion by atom63** [nuxt/nuxt#25124] [Questions]
Hello Vue/Nuxt community,

I recently migrated my static website from Nuxt 2 to Nuxt 3 and have encountered a performance issue specifically on mobile devices. The animations on the website have become noticeably slower, and DOM elements are disappearing when they are out of the viewport. This issue is not present in the desktop version or the previous Nuxt 2 version of the site.

**Live Sites:**

Nuxt2: https://atom63-id2mencoq-atom63.vercel.app/

Nuxt3: https://atom63-nuxt3.vercel.app/

---

**Issue Description:**

**Environment:**
The issue occurs on mobile browsers (tested on Edge, Chrome, Safari).

**Symptoms:**
Animations across the site are sluggish and not smooth.
DOM elements disappear when scrolled out of the viewport and flickering in when scrolled into the viewport.

**Expected Behavior:** 
Smooth animations and consistent DOM element visibility, similar to the desktop version and the old Nuxt 2 site.

---

**Steps Taken:**
- Ensured all dependencies, especially animation and CSS-related, are updated and compatible with Nuxt 3. Deployment is error free.
- Image optimized with Photoshop save for web.
- Tried to add `transform: translate3d(0, 0, 0);` to enable hardware acceleration. Some animation seems running smoother but still laggy comparing to the Nuxt 2 site.
- Tested with and without GSAP SmoothScroll and AOS, nothing was effectively improved.

---

**Additional Notes:**

Both sites used similar packges like TailwindCSS, GSAP SmoothScroller, Swiper, AOS. The performance on mobile is dramatically different.

Packages installed on Nuxt 3 site:

```
"dependencies": {
    "@headlessui/vue": "^1.7.17",
    "@nuxt/image": "^1.1.0",
    "@nuxtjs/google-fonts": "^3.1.3",
    "@nuxtjs/i18n": "8.0.0-rc.3",
    "@pinia/nuxt": "0.4.11",
    "@tailwindcss/aspect-ratio": "0.4.2",
    "@tailwindcss/forms": "0.5.6",
    "@tailwindcss/typography": "0.5.9",
    "@unocss/preset-icons": "^0.58.3",
    "@vercel/speed-insights": "^1.0.2",
    "fast-average-color": "^9.4.0",
    "gsap": "^3.12.4",
    "nuxt-lenis": "^1.0.4",
    "nuxt-swiper": "^1.2.2",
    "nuxt-video-player": "^1.0.17",
    "pinia": "2.1.6",
    "vue-responsive-video-background-player": "^2.3.1",
    "vue-virtual-scroller": "2.0.0-beta.8"
  },
  "devDependencies": {
    "@antfu/eslint-config": "0.41.0",
    "@iconify-json/carbon": "1.1.20",
    "@iconify-json/ic": "1.1.14",
    "@iconify-json/ph": "1.1.6",
    "@iconify-json/twemoji": "1.1.12",
    "@iconify/json": "^2.2.165",
    "@nuxt/devtools": "^0.8.2",
    "@nuxtjs/color-mode": "3.3.0",
    "@nuxtjs/tailwindcss": "6.8.0",
    "@unocss/nuxt": "0.55.3",
    "@vueuse/nuxt": "10.4.1",
    "daisyui": "^4.5.0",
    "eslint": "8.48.0",
    "eslint-plugin-vue": "9.17.0",
    "nuxt": "3.7.0",
    "nuxt-gtag": "^1.1.2",
    "nuxt-marquee": "^1.0.0",
    "nuxt-vercel-analytics": "^1.0.0",
    "nuxt3-aos": "^1.0.18",
    "sass": "1.66.1",
    "sass-loader": "13.3.2",
    "typescript": "5.2.2"
  }
```

Packages installed on Nuxt 2 site:

```
 "dependencies": {
    "@nuxtjs/universal-storage": "^0.5.9",
    "@vercel/analytics": "^1.1.1",
    "@vercel/speed-insights": "^1.0.2",
    "aos": "^2.3.4",
    "core-js": "^3.25.3",
    "fast-average-color": "^9.4.0",
    "gsap": "npm:@gsap/shockingly@^3.12.2",
    "imagesloaded": "^5.0.0",
    "isotope-layout": "^3.0.6",
    "keen-ui": "^1.4.0",
    "lodash": "^4.17.21",
    "matter-js": "^0.19.0",
    "nuxt": "^2.15.8",
    "p5": "^1.7.0",
    "swiper": "^5.4.5",
    "three": "^0.156.1",
    "typewriter-effect": "^2.21.0",
    "v-lazy-image": "^2.1.1",
    "vue": "^2.7.10",
    "vue-awesome-swiper": "^4.1.1",
    "vue-images-loaded": "^1.1.2",
    "vue-lazyload-video": "^0.1.15",
    "vue-marquee-text-component": "^1.2.0",
    "vue-server-renderer": "^2.7.10",
    "vue-template-compiler": "^2.7.10",
    "vue-unicons": "^3.3.1",
    "vueisotope": "^3.1.2"
  },
  "devDependencies": {
    "@iconify/vue2": "^2.1.0",
    "@nuxt/image": "^0.7.1",
    "@nuxtjs/google-fonts": "^3.0.2",
    "autoprefixer": "^10.4.15",
    "locomotive-scroll": "^4.1.4",
    "node-sass": "^6.0.1",
    "nuxt-gsap-module": "^1.7.2",
    "postcss": "^8.4.28",
    "sass": "^1.66.1",
    "sass-loader": "^10.4.1",
    "tailwindcss": "^3.3.3"
  }
```

--- 

**Queries:**

- Are there known issues with Nuxt 3 concerning mobile performance, especially around animations and DOM rendering?
- Are there any recommended practices or configurations in Nuxt 3 for optimizing mobile performance?

---

**Last Note:**

I appreciate any insights or suggestions the community can offer. If additional information is needed, I'm happy to provide it.

P.S. My background is not in development, so please bear with me if I'm asking any beginner questions. 

Thank you!

**Comment by huang-julien** [comment]
Hi :wave: do you have a small reproduction ? It is usefull to actually see what's going on. 

Animations are quite tricky, that's probably not using pure CSS. Maybe it's missing a `requestAnimationFrame` or that a watcher is triggered too much somewhere
